### PR TITLE
APB-444 - Play Upgrade refactor 

### PR DESCRIPTION
**Pull request by tev1622**

### CODE REVIEW

File: app/uk/gov/hmrc/agentclientauthorisation/connectors/AuthoriseConnector.scala [hunk 1]
```diff
@@ -25,16 +26,15 @@ import uk.gov.hmrc.play.http.{HeaderCarrier, HttpGet, HttpResponse}
...
+    val url = s"$baseUrl/authorise/read/mtd-sa/${mtdClientId.value}?confidenceLevel=50&delegatedAuthRule=mtd-sa-auth"
+    httpGet.GET[HttpResponse](url) map { _ => true } recover {
       case e: Throwable => Logger.info(s"Authorise call threw exception ${e.getMessage}", e)
-      false
+        false
     }
   }
-
-
-  def buildUrl(mtdClientId: MtdClientId): URL = {
```
Comment (frankoid) [inline app/uk/gov/hmrc/agentclientauthorisation/connectors/AuthoriseConnector.scala]: how come `buildUrl` has been inlined?

File: app/uk/gov/hmrc/agentclientauthorisation/frontendWiring.scala [hunk 2]
```diff
@@ -16,21 +16,14 @@
...
-import uk.gov.hmrc.agentclientauthorisation.service.{ClientService, InvitationsService, SessionStoreService}
 import uk.gov.hmrc.http.cache.client.SessionCache
 import uk.gov.hmrc.play.audit.http.config.LoadAuditingConfig
 import uk.gov.hmrc.play.audit.http.connector.{AuditConnector => Auditing}
 import uk.gov.hmrc.play.config.{AppName, RunMode, ServicesConfig}
 import uk.gov.hmrc.play.frontend.auth.connectors.AuthConnector
 import uk.gov.hmrc.play.http.ws.{WSDelete, WSGet, WSPost, WSPut}
 
-object FrontendAuditConnector extends Auditing with AppName with RunMode {
+object  FrontendAuditConnector extends Auditing with AppName with RunMode {
```
Comment (frankoid) [inline app/uk/gov/hmrc/agentclientauthorisation/frontendWiring.scala]: extra space added here

File: app/uk/gov/hmrc/agentclientauthorisation/service/ClientService.scala [hunk 3]
```diff
@@ -24,7 +26,7 @@ import uk.gov.hmrc.play.http.HeaderCarrier
 
 import scala.concurrent.Future
 
-class ClientService(agenciesFakeConnector: AgenciesFakeConnector) {
+class ClientService @Inject() (agenciesFakeConnector: AgenciesFakeConnector) {
```
Comment (frankoid) [inline app/uk/gov/hmrc/agentclientauthorisation/service/ClientService.scala]: is it deliberate that there is no `@Singleton` annotation?

https://github.com/google/guice/wiki/Scopes

File: app/uk/gov/hmrc/agentclientauthorisation/frontendWiring.scala [hunk 4]
```diff
@@ -53,24 +47,24 @@ object AgentClientAuthorisationSessionCache extends SessionCache with AppName wi
 trait ServiceRegistry extends ServicesConfig {
```
Comment (frankoid) [inline app/uk/gov/hmrc/agentclientauthorisation/frontendWiring.scala]: best to delete commented out code - we can always recover it from git if needed.

In fact are `ServiceRegistry` and `ControllerRegistry` needed at all now we're using Guice?

**Comment by frankoid** [comment]
Looks like a good change overall, I've left comments on some specific parts.

File: app/uk/gov/hmrc/agentclientauthorisation/connectors/AuthoriseConnector.scala (hunk 1 above)
Comment (tev1622) [inline app/uk/gov/hmrc/agentclientauthorisation/connectors/AuthoriseConnector.scala]: there is no need for the `buildUrl`, `val url = ...` accomplishes the same thing

Comment (frankoid) [inline app/uk/gov/hmrc/agentclientauthorisation/connectors/AuthoriseConnector.scala]: yes, but it's good for functions to be short and have one/few responsibilities

**Pull request merged**